src/appkit.rs

```diff
@@ -1,57 +1,70 @@
-use crate::{CAMetalLayer, Layer};
 use core::ffi::c_void;
-use core_graphics::{base::CGFloat, geometry::CGRect};
-use objc::{
-    msg_send,
-    runtime::{BOOL, YES},
-};
+use objc2::rc::Retained;
+use objc2::ClassType;
+use objc2_foundation::{NSObject, NSObjectProtocol};
+use objc2_quartz_core::CAMetalLayer;
 use raw_window_handle::AppKitWindowHandle;
 use std::ptr::NonNull;
 
+use crate::Layer;
+
+/// Get or create a new [`Layer`] associated with the given
+/// [`AppKitWindowHandle`].
+///
+/// # Safety
 ///
+/// The handle must be valid.
 pub unsafe fn metal_layer_from_handle(handle: AppKitWindowHandle) -> Layer {
-    metal_layer_from_ns_view(handle.ns_view)
+    unsafe { metal_layer_from_ns_view(handle.ns_view) }
 }
 
+/// Get or create a new [`Layer`] associated with the given `NSView`.
 ///
+/// # Safety
+///
+/// The view must be a valid instance of `NSView`.
 pub unsafe fn metal_layer_from_ns_view(view: NonNull<c_void>) -> Layer {
-    let view: cocoa::base::id = view.cast().as_ptr();
+    // SAFETY: Caller ensures that the view is valid.
+    let obj = unsafe { view.cast::<NSObject>().as_ref() };
```

Comment:
Could we already cast to `NSView` here, noting that the caller guarantees it's an `NSView`, or does that mess with calling `is_kind_of()`?

Reply:
We could, I think it was in there in case the user didn't actually provide a `NSView`, but instead provided a layer directly. I've fixed this in https://github.com/rust-windowing/raw-window-metal/pull/19 anyhow by introducing `Layer::from_layer`.